packages/itwinui-react/src/core/Tile/Tile.tsx

```diff
@@ -147,239 +105,463 @@ type TileProps = {
+// Tile.BadgeContainer component
+
+const TileBadgeContainer = polymorphic.div(
+  'iui-tile-thumbnail-badge-container',
 );
+TileBadgeContainer.displayName = 'Tile.BadgeContainer';
 
-type TitleIconProps = {
-  isLoading?: boolean;
-  isSelected?: boolean;
-  isNew?: boolean;
-  status?: 'positive' | 'warning' | 'negative';
+// ----------------------------------------------------------------------------
+// Tile.Name component
+type TileNameOwnProps = {
+  name?: string;
 };
 
-const TitleIcon = ({
-  isLoading = false,
-  isSelected = false,
-  isNew = false,
-  status,
-}: TitleIconProps) => {
+const TileName = React.forwardRef((props, ref) => {
+  const { className, children, name, ...rest } = props;
+  return (
+    <Box className={cx('iui-tile-name', className)} ref={ref} {...rest}>
+      {children ?? name}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div', TileNameOwnProps>;
+TileBadgeContainer.displayName = 'Tile.Name';
+
+// ----------------------------------------------------------------------------
+// Tile.NameIcon component
+
+const TileNameIcon = React.forwardRef((props, ref) => {
+  const { children, ...rest } = props;
+  const { status, isLoading, isSelected, isNew } = useSafeContext(TileContext);
+
   const StatusIcon = !!status && StatusIconMap[status];
+  let icon;
 
+  if (StatusIcon) {
+    icon = <StatusIcon aria-hidden />;
+  }
   if (isLoading) {
-    return (
-      <ProgressRadial
-        className='iui-tile-status-icon'
-        aria-hidden
-        indeterminate
-      />
+    icon = (
+      <ProgressRadial style={{ height: '100%' }} aria-hidden indeterminate />
     );
   }
   if (isSelected) {
-    return <SvgCheckmark className='iui-tile-status-icon' aria-hidden />;
+    icon = <SvgCheckmark aria-hidden />;
   }
   if (isNew) {
-    return <SvgNew className='iui-tile-status-icon' aria-hidden />;
+    icon = <SvgNew aria-hidden />;
   }
-  if (StatusIcon) {
-    return <StatusIcon className='iui-tile-status-icon' />;
-  }
-  return null;
+
+  return children || icon ? (
+    <Box className='iui-tile-status-icon' ref={ref} {...rest}>
+      {children ?? icon}
+    </Box>
+  ) : null;
+}) as PolymorphicForwardRefComponent<'div'>;
+TileNameIcon.displayName = 'Tile.NameIcon';
+
+// ----------------------------------------------------------------------------
+// Tile.NameLabel component
+
+const TileNameLabel = React.forwardRef((props, ref) => {
+  const { children, ...rest } = props;
+  const { isActionable, isDisabled, onClick } = useSafeContext(TileContext);
+
+  return (
+    <Box as='span' className='iui-tile-name-label' ref={ref} {...rest}>
+      {isActionable && onClick ? (
+        <LinkAction
+          as='button'
+          onClick={!isDisabled ? onClick : undefined}
+          aria-disabled={isDisabled}
+        >
+          {children}
+        </LinkAction>
+      ) : (
+        children
+      )}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div'>;
```

Comment:
This component should be forwarded as a `span` instead of a `div`

Reply:
Done